modules/auxiliary/scanner/http/git_scanner.rb

```diff
@@ -0,0 +1,131 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Auxiliary
+
+  # Exploit mixins should be called first
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::WmapScanServer
+  # Scanner mixin should be near last
+  include Msf::Auxiliary::Scanner
+  include Msf::Auxiliary::Report
+
+  def initialize
+    super(
+      'Name'        => 'HTTP Git Scanner',
+      'Description' => 'Detect git directories and files and analize its content.',
+      'Author'       => ['t0nyhj'],
+      'License'     => MSF_LICENSE
+    )
+
+    register_options(
+      [
+        OptString.new('PATH', [ true,  "The test path to .git directory", '/'])#,
+        #OptBool.new('GET_SOURCE', [ false, "Attempt to obtain file source code", true ]),
+        #OptBool.new('SHOW_SOURCE', [ false, "Show source code", true ])
+
+      ], self.class)
+
+    register_advanced_options(
+      [
+        OptInt.new('ErrorCode', [ true, "Error code for non existent directory", 404]),
+        OptPath.new('HTTP404Sigs',   [ false, "Path of 404 signatures to use",
+            File.join(Msf::Config.data_directory, "wmap", "wmap_404s.txt")
+          ]
+        ),
+        OptBool.new('NoDetailMessages', [ false, "Do not display detailed test messages", true ])
+
+      ], self.class)
+  end
+
+  def run_host(target_host)
+    conn = true
+    ecode = nil
+    emesg = nil
+
+    word1 = 'core'
+    word2 = 'remote'
+    word3 = 'branch'
+
+    tpath = normalize_uri(datastore['PATH'])
+    if tpath[-1,1] != '/'
```

Comment:
This shouldn't be necessary, because normalize_uri should do that for you already.